### PR TITLE
Implementing JSON Web Token Library for User Authentication

### PR DESCRIPTION
**Pull request by hanwenzhang123**
## POC - Working In Progress
- Application is NOT break
- Build is successful

## What I did
- Implement JWT Util and Filter classes
- General Random JWT Token and send the token to the client

## TO DO
- [ ] TOO MANY, the functionality is not working

### CODE REVIEW

File: code/SoloSavingsApp/src/main/resources/application.properties [hunk 1]
```diff
@@ -10,4 +10,7 @@ spring.jpa.show-sql: true
 
 server.port=8888
 server.error.path=/error
-logging.level.org.springframework.security=TRACE
\ No newline at end of file
+logging.level.org.springframework.security=TRACE
+
+jwt.secret=mysecretkey
```
Comment (WAftring) [inline code/SoloSavingsApp/src/main/resources/application.properties:15]: Going forward we probably need another way to manage this secret. But that is likely after initial JWT implementation.

**Comment by WAftring** [comment]
JWT now works but we need to restructure how we handle the authorization token.

Only API calls should require authentication which means tinkering with the AJAX calls to ensure they are pulling from the newly created JWT token cookie.

Lots more work to be done but getting there...

**Comment by hanwenzhang123** [comment]
@WAftring, Great job on the session!!!
Screenshot 2023-10-02 at 8 04 43 PM
I see what you mean by there is still a lot needs to be done...

**Comment by WAftring** [comment]
This PR is an absolute **monster**. @hanwenzhang123 @to-whitney let's schedule some time to go over this.

File: code/SoloSavingsApp/src/main/java/com/soloSavings/controller/TransactionController.java [hunk 2]
```diff
@@ -36,44 +40,58 @@ public class TransactionController {
     //Income
     @Autowired
     TransactionService transactionServiceImpl;
+    @Autowired
+    SecurityContext securityContext;
 
     @RequestMapping(value = "/add/{id}", method = RequestMethod.POST)
```
Comment (WAftring) [inline code/SoloSavingsApp/src/main/java/com/soloSavings/controller/TransactionController.java:46]: Remove id from request mapping

File: code/database/init-database.sql
Comment (WAftring) [inline code/database/init-database.sql:1]: We need to have the username be the unique field instead of the email.

**Comment by WAftring** [comment]
Need to update the unit tests.

**Review by WAftring — APPROVED** [review]

**Pull request merged**